spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/TestMetadataTableUUIDMetrics.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.source;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.UUID;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+@Tag("spark")
+public class TestMetadataTableUUIDMetrics {
+
+  private static SparkSession spark;
+
+  @TempDir static Path warehouse;
+
+  @BeforeAll
+  static void startSpark() {
+    spark =
+        SparkSession.builder()
+            .master("local[2]")
+            .appName("uuid-readable-metrics-test")
+            .config("spark.ui.enabled", "false")
+            .config("spark.sql.catalog.test", "org.apache.iceberg.spark.SparkCatalog")
+            .config("spark.sql.catalog.test.type", "hadoop")
+            .config("spark.sql.catalog.test.warehouse", warehouse.toString())
+            .getOrCreate();
+  }
+
+  @AfterAll
+  static void stopSpark() {
+    if (spark != null) {
+      spark.stop();
+    }
+  }
+
+  @Test
+  public void testUUIDMetricsReadableInAllFiles() {
```

Comment:
we already have `TestMetadataTables`, so you can just add this test there (no need to create a separate test class)